docs/dev/guidelines/database.rst

```diff
@@ -0,0 +1,236 @@
+
+
+    With ordered collections, you have to be very careful with the way you persist the objects in the database. You must first persist the child object without a relation to the parent object. Then, you recreate the association and persist the parent object. Example of how to correctly persist objects in an ordered collection:
+
+       .. code-block:: java
+
+        // ProgrammingAssessmentService
+        List<Feedback> savedFeedbacks = new ArrayList<>();
+        result.getFeedbacks().forEach(feedback -> {
+           // cut association to parent object
+           feedback.setResult(null);
+           // persist the child object without an association to the parent object. IMPORTANT: Use the object returned from the database!
+           feedback = feedbackRepository.save(feedback);
+           // restore the association to the parent object
+           feedback.setResult(result);
+           savedFeedbacks.add(feedback);
+       });
+       
+
+        // set the association of the parent to its child objects which are now persisted in the database
+        result.setFeedbacks(savedFeedbacks);
+        // persist the parent object
+        return resultRepository.save(result);
+
+
+Solutions for known issues
+==========================
+
+* ``org.hibernate.LazyInitializationException : could not initialize proxy – no Session`` caused by ``fetchType.LAZY``. You must explicitly load the associated object from the database before trying to access those. Example of how to eagerly fetch the feedbacks with the result:
+
+ .. code-block:: java
+
+    // ResultRepository.java
+    @Query("select r from Result r left join fetch r.feedbacks where r.id = :resultId")
+    Optional<Result> findByIdWithEagerFeedbacks(@Param("resultId") Long id);
+
+
+* ``JpaSystemException: null index column for collection`` caused by ``@OrderColumn`` annotation:
+
+ #. Save the child entity (e.g., `Feedback <https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/domain/Feedback.java>`_) without connection to the parent entity (e.g., `Result <https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/domain/Result.java>`_)
+ #. Add back the connection of the child entity to the parent entity.
+ #. Save the parent entity.
+ #. Always use the returned value after saving the entity, see: ``feedback = feedbackRepository.save(feedback);``
+
+ .. code:: java
+
+    // ProgrammingAssessmentService
+    List<Feedback> savedFeedbacks = new ArrayList<>();
+    result.getFeedbacks().forEach(feedback -> {
+        // cut association to parent object
+        feedback.setResult(null);
+        // persist the child object without an association to the parent object. IMPORTANT: Use the object returned from the database!
+        feedback = feedbackRepository.save(feedback);
+        // restore the association to the parent object
+        feedback.setResult(result);
+        savedFeedbacks.add(feedback);
+    });
```

Comment:
This block seems to be duplicated. This is already explained above. You could just reference the ordered collections section.

Reply:
See e56ec3f